src/java/com/rapleaf/jack/queries/Record.java

```diff
@@ -72,6 +77,74 @@ public Boolean getBoolean(Column column) {
     return value == null ? null : (Boolean)value;
   }
 
+  public <A extends AttributesWithId, M extends ModelWithId> A getAttributes(Table<A, M> tableType) {
+    String tableName = tableType.getAlias();
+    Constructor<A> constructor;
+    try {
+      constructor = tableType.getAttributesType().getConstructor(Long.TYPE);
+    } catch (NoSuchMethodException e) {
+      throw new RuntimeException(e);
+    }
+
+    if (constructor == null || tableName == null) {
+      throw new RuntimeException("No columns from Table " + tableName + " are included in this record");
+    }
+
+    Long id = null;
+    Map<Enum, Object> fieldMap = Maps.newHashMap();
+    for (Map.Entry<Column, Object> entry : columns.entrySet()) {
+      Column column = entry.getKey();
+      if (column.getTable().equals(tableName)) {
+        if (column.getField() != null) {
+          fieldMap.put(column.getField(), entry.getValue());
+        } else if (id == null) {
+          id = ((Number)(entry.getValue())).longValue();
+        } else {
+          throw new RuntimeException("The record contains multiple IDs for Table " + tableName);
```

Comment:
so I'm not sure how fixable it is, but this was a bit of a hack at the time.  column.getField() is null for ID fields, so I guess that it's the ID.  but the problem is that if your Record has columns from multiple tables, it can't figure out which ID is which.  it could also get the entirely wrong ID if you selectAttributes for a record type which doesn't correspond to the ID present.

I think this is still useful to have whether or not it's fixed, just FYI in case you see an easy fix.